core/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -184,4 +190,27 @@ private void failIfOverShardCountLimit(ClusterService clusterService, int shardC
         }
     }
 
+    private void resolveIndexBoost(SearchRequest searchRequest, ClusterState clusterState) {
+        if (searchRequest.source() == null) {
+            return;
+        }
+
+        List<Tuple<String, Float>> indexBoost = searchRequest.source().indexBoost();
+        if (indexBoost == null) {
+            return;
+        }
+
+        List<Tuple<String, Float>> concreteIndexBoost = new ArrayList<>();
+        for (Tuple<String, Float> ib : indexBoost) {
+            String[] concreteIndices =
+                indexNameExpressionResolver.concreteIndexNames(clusterState, IndicesOptions.lenientExpandOpen(), ib.v1());
+
+            for (String concreteIndex : concreteIndices) {
+                concreteIndexBoost.add(new Tuple<>(concreteIndex, ib.v2()));
+            }
+        }
+
+        searchRequest.source().indexBoost(concreteIndexBoost);
```

Comment:
I find it odd that we modify the original source builder with the resolved indices names and replace what we originally had. Would it be possible to transport the resolved indices differently? I think we should serialize this new info separately and carefully handle bw compatibility around that.